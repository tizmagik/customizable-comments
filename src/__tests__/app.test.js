const { Application } = require("probot");
const bot = require("../");
const payload = require("./fixtures/pull_request.opened.json");

jest.mock("probot-config");

describe("probot-cc", () => {
  let app;
  let github;

  beforeEach(() => {
    app = new Application();
    app.load(bot);
    github = {
      issues: {
        createComment: jest.fn().mockReturnValue(Promise.resolve())
      }
    };
    app.auth = () => Promise.resolve(github);
  });

  describe("on pull_request.opened", () => {
    it("creates comment with preview sandbox helpers", async () => {
      await app.receive(payload);

      expect(github.issues.createComment).toHaveBeenCalled();
      expect(github.issues.createComment.mock.calls[0][0]).toMatchSnapshot();
    });
  });
});
