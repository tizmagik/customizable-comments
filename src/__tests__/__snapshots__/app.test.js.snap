// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probot-cc on pull_request.opened creates comment with preview sandbox helpers 1`] = `
Object {
  "body": "_You have succesfully installed probot-cc, now please create a \`.github/customizable-comments.yml\` file in your repository to customize this message._",
  "number": 1,
  "owner": "tizmagik",
  "repo": "testing-pr-opened",
}
`;
